pycaching/trackable.py

```diff
@@ -17,7 +17,10 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
         except AttributeError:
             logging.debug("Lazy loading: %s", func.__name__)
-            self.geocaching.load_trackable(self.tid, self)
+            try:
+                self.geocaching.load_trackable_by_url(self.trackable_page, self)
+            except AttributeError:
+                self.geocaching.load_trackable(self.tid, self)
```

Comment:
Regarding this, I see two possible solutions...
1. Catch the infinite recursion exception and raise a ValueError
2. Using has_attr(self, '_tid') to determine if the _tid property exist and if that doesn't exist raise ValueError

Personally I think I would prefer 2 since it feels bad allowing an endless recursion but using hasattr() seems to be considered unpythonic and thus 1 is to be preferred?

What's you opinion?